lib/benchee/conversion/format.ex

```diff
@@ -6,71 +6,73 @@ defmodule Benchee.Conversion.Format do
   See `Benchee.Conversion.Count` and `Benchee.Conversion.Duration` for examples
   """
 
+  alias Benchee.Conversion.Unit
+
   @doc """
   Formats a number as a string, with a unit label. See `Benchee.Conversion.Count`
   and `Benchee.Conversion.Duration` for examples
   """
   @callback format(number) :: String.t
 
-  @doc """
-  The label for a given unit, as a String.  See `Benchee.Conversion.Count`
-  and `Benchee.Conversion.Duration` for examples
-  """
-  @callback label(Benchee.Conversion.Scale.unit) :: String.t
-
-  @doc """
-  A string that appears between a value and a unit label when formatted as a
-  String. For example, a space: `5.67 M` or an empty string: `5.67M`
-  """
-  @callback separator :: String.t
-
   # Generic formatting functions
 
   @doc """
   Formats a unit value with specified label and separator
   """
-  def format({count, _unit}, label, separator) do
+  def format(count, label, separator) do
     separator = separator(label, separator)
     "~.#{float_precision(count)}f~ts~ts"
     |> :io_lib.format([count, separator, label])
     |> to_string
   end
 
   @doc """
-  Formats a unit value with the label and separator supplied by `module`. The
-  specified module should provide `label/1` and `separator/0` functions
+  Formats a unit value in the domain described by `module`. The module should
+  provide a `units/0` function that returns a Map like
+
+      %{ :unit_name => %Benchee.Conversion.Unit{ ... } }
+
+  Additionally, `module` may specify a `separator/0` function, which provides a
+  custom separator string that will appear between the value and label in the
+  formatted output. If no `separator/0` function exists, the default separator
+  (a single space) will be used.
   """
   def format({count, unit}, module) do
-    format({count, unit}, module.label(unit), module.separator)
+    format(count, label(module, unit), separator(module))
   end
 
   @doc """
-  Scales a number to the most appropriate unit, and formats the scaled value
-  with the label and separator supplied by `module`. The
-  specified module should provide `label/1` and `separator/0` functions
+  Scales a number to the most appropriate unit as defined in `module`, and
+  formats the scaled value with a label. The module should provide a `units/0`
+  function that returns a Map like
+
+      %{ :unit_name => %Benchee.Conversion.Unit{ ... } }
   """
   def format(number, module) do
     number
     |> module.scale
     |> format(module)
   end
 
-  @doc """
-  Fetches a label from a map of units
-  """
-  def label(units, unit) do
-    units
-    |> Map.fetch!(unit)
-    |> Map.fetch!(:short)
+  # Returns the separator defined in `module.separator/0`, or the default, a space
+  defp separator(module) do
+    case function_exported?(module, :separator, 0) do
+      true  -> module.separator()
```

Comment:
unimportant thing, I usually like to stick to the ruby way of things without parentheses for method calls without args but nothing that needs to change :)